src/azure-cli/azure/cli/command_modules/synapse/generated/_params.py

```diff
@@ -0,0 +1,956 @@
+
+    with self.argument_context('synapse ip-firewall-rule update') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('workspace_name', type=str, help='The name of the workspace', id_part='name')
+        c.argument('rule_name', type=str, help='The IP firewall rule name', id_part='child_name_1')
+        c.argument('end_ip_address', type=str, help='The end IP address of the firewall rule. Must be IPv4 format. '
+                   'Must be greater than or equal to startIpAddress')
+        c.argument('start_ip_address', type=str,
+                   help='The start IP address of the firewall rule. Must be IPv4 format')
+
+    with self.argument_context('synapse ip-firewall-rule delete') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('workspace_name', type=str, help='The name of the workspace', id_part='name')
+        c.argument('rule_name', type=str, help='The IP firewall rule name', id_part='child_name_1')
+
+    with self.argument_context('synapse ip-firewall-rule replace-all') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('workspace_name', type=str, help='The name of the workspace', id_part='name')
+        c.argument('ip_firewall_rules', type=validate_file_or_dict, help='IP firewall rule properties Expected value: '
+                   'json-string/@json-file.')
+
+    with self.argument_context('synapse ip-firewall-rule wait') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('workspace_name', type=str, help='The name of the workspace', id_part='name')
+        c.argument('rule_name', type=str, help='The IP firewall rule name', id_part='child_name_1')
+
+    with self.argument_context('synapse sql-pool list') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('workspace_name', type=str, help='The name of the workspace')
+
+    with self.argument_context('synapse sql-pool show') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('workspace_name', type=str, help='The name of the workspace', id_part='name')
+        c.argument('sql_pool_name', options_list=['--name', '-n', '--sql-pool-name'], type=str, help='SQL pool name',
+                   id_part='child_name_1')
+
+    with self.argument_context('synapse sql-pool create') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('workspace_name', type=str, help='The name of the workspace')
+        c.argument('sql_pool_name', options_list=['--name', '-n', '--sql-pool-name'], type=str, help='SQL pool name')
+        c.argument('tags', tags_type)
+        c.argument('location', arg_type=get_location_type(self.cli_ctx),
+                   validator=get_default_location_from_resource_group)
+        c.argument('sku', action=AddSku, nargs='*', help='SQL pool SKU')
```

Comment:
Usually you only need to provide the required option (`name` or `tier`, in synapse `name` is required for [sku](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/synapse/common/v1/types.json#L20-L36)) in sku, I saw some previous code using `--sku` as option, it takes in `sku.name` values. It makes the interface simpler. This could be open for discussion.